282: Style communities

bronzehedwick opened the pull request:
This PR accomplishes the following.

1. Styles the `/community` page.
2. Style accommodates missing images. 
3. Removes the "Add community" button from `/community`.
4. Truncate long community names on `/community`.
5. Refactor button styles to a mixin, so they can be applied to arbitrary elements (this was useful here because adding a `.button` class was difficult in this context).

Re #282

bronzehedwick committed:
282: Style communities

bronzehedwick committed:
282: Remove "Add community" button

bronzehedwick committed:
282: Truncate long community titles

On summary pages, like `/community`, truncate the length of the title,
as the design constrains the height of each item.

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr765-cldcwamqw3gho38y9egacktyobyuktld.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44b6c8986000a9f591830

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr765-cldcwamqw3gho38y9egacktyobyuktld.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44b6c8986000a9f591830

michael-wynne-wsu commented:
@bronzehedwick 

One issue: I don't like that with 1 or 2 communities, on a larger display, the columns stretch - I think keeping the proportions would be preferable.

One question: what's the target thumbnail proportion? (just curious for documentation purposes)

michael-wynne-wsu commented:
Ah I forgot the screenshots!

![Screenshot 2025-03-26 at 12 50 13 PM](https://github.com/user-attachments/assets/e9fb6db4-9711-47f4-87ae-8fc7e8da0047)

![Screenshot 2025-03-26 at 12 48 18 PM](https://github.com/user-attachments/assets/5eeb01a0-651c-432d-a6fb-d6d0808e853b)

![Screenshot 2025-03-26 at 12 47 25 PM](https://github.com/user-attachments/assets/0218afc1-f0f7-433c-a1fb-617b25a09699)

michael-wynne-wsu commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr765-cldcwamqw3gho38y9egacktyobyuktld.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/67e44b6c8986000a9f591830

bronzehedwick committed:
282: Keep community image proportions

Have a max width instead of stretching to fill available space. Correct
the max width value.

bronzehedwick commented:
> > 
> 
> The target is `288px`. I'm attempting to get close to the v3 implementation, but tweaking a little. v3 is `286.8px`, so I rounded up to avoid subpixel rounding, and then added another `1px` so it was an even `19rem`.

Oops, made a mistake in my explanation @michael-wynne-wsu! The numbers I was giving you, above, are for the *height*, not the width. 

The width is a similar story. On v3, it's `382.4px`, so in v4, I made the ideal width `382px`, again, to avoid subpixel rounding.

bronzehedwick commented:
Pushed a fix for the image issue @michael-wynne-wsu! Let me know if that looks better.

michael-wynne-wsu reviewed (approved):
Perfect, good to go!
